### PR TITLE
feat: setSelection method for editable text components

### PR DESCRIPTION
**Pull request by farfromrefug**
this add support for `setSection` for all editable text components

### CODE REVIEW

File: packages/core/ui/editable-text-base/index.ios.ts [hunk 1]
```diff
@@ -189,6 +190,17 @@ export abstract class EditableTextBase extends EditableTextBaseCommon {
 
 		this.nativeTextViewProtected.autocorrectionType = newValue;
 	}
+	public setSelection(start: number, stop?: number) {
+		const view = this.nativeTextViewProtected;
+		if (stop !== undefined) {
+			const begin = view.beginningOfDocument;
+			view.selectedTextRange = view.textRangeFromPositionToPosition(view.positionFromPositionOffset(begin, start), view.positionFromPositionOffset(begin, stop));
```
Comment (rigor789) [inline packages/core/ui/editable-text-base/index.ios.ts:197]: Perhaps this makes this line a bit more readable
```suggestion
			view.selectedTextRange = view.textRangeFromPositionToPosition(
						view.positionFromPositionOffset(begin, start),
						view.positionFromPositionOffset(begin, stop)
			);
```

**Review by rigor789 — APPROVED** [review]

File: packages/core/ui/editable-text-base/index.ios.ts [hunk 2]
```diff
@@ -189,6 +190,20 @@ export abstract class EditableTextBase extends EditableTextBaseCommon {
 
 		this.nativeTextViewProtected.autocorrectionType = newValue;
 	}
+	public setSelection(start: number, stop?: number) {
+		const view = this.nativeTextViewProtected;
+		if (stop !== undefined) {
```
Comment (NathanWalker) [inline packages/core/ui/editable-text-base/index.ios.ts:195]: Might be good to wrap all this in conditional to protect against view (android side does it already but always best to be careful with ios as well):
```
if (view) {
```

**Review by NathanWalker — CHANGES_REQUESTED** [review]

**Comment by farfromrefug** [comment]
@NathanWalker not sure why it says changes are requested still. I did the asked changes

File: packages/core/ui/editable-text-base/index.ios.ts [hunk 3]
```diff
@@ -189,6 +190,22 @@ export abstract class EditableTextBase extends EditableTextBaseCommon {
...
+				const begin = view.beginningOfDocument;
+				view.selectedTextRange = view.textRangeFromPositionToPosition(
+					view.positionFromPositionOffset(begin, start), 
+					view.positionFromPositionOffset(begin, stop)
+				);
+			} else {
+				const begin = view.beginningOfDocument;
+				const pos = view.positionFromPositionOffset(begin, start);
+				view.selectedTextRange = view.textRangeFromPositionToPosition(pos, pos);
+			}
```
Comment (rigor789) [inline packages/core/ui/editable-text-base/index.ios.ts:206]: Formatting seems to be off here - did you run `npm run setup` after cloning the repo to ensure husky & git-hooks are installed? These should then auto-format changed files on commit.

**Review by rigor789 — DISMISSED** [review]
LGTM - couple formatting issues only.

File: packages/core/ui/editable-text-base/index.ios.ts (hunk 3 above)
Comment (farfromrefug) [reply packages/core/ui/editable-text-base/index.ios.ts:206]: yes I do and I use eslint. wonder why it does not work. shouldn't it prevent the commit or auto format ?

**Review by rigor789 — APPROVED** [review]

**Review by NathanWalker — APPROVED** [review]

**Pull request merged**